File: app/src/main/java/igrek/forceawaken/ui/input/TextAddedListener.java
package igrek.forceawaken.ui.input;

import android.text.Editable;
import android.text.TextWatcher;

public abstract class TextAddedListener implements TextWatcher {
	
	private int beforeLength;
	
	@Override
	public void beforeTextChanged(CharSequence s, int start, int count, int after) {
		beforeLength = s.length();
	}
	
	@Override
	public void onTextChanged(CharSequence s, int start, int before, int count) {
	}
	
	@Override
	public void afterTextChanged(Editable s) {
		if (s.length() > beforeLength) {
			onTextAdded(s.toString());
		}
	}
	
	protected abstract void onTextAdded(String newValue);
}

File: app/src/main/java/igrek/forceawaken/layout/input/TextAddedListener.kt
package igrek.forceawaken.layout.input

import android.text.Editable
import android.text.TextWatcher

abstract class TextAddedListener : TextWatcher {
    private var beforeLength = 0
    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
        beforeLength = s.length
    }

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
    override fun afterTextChanged(s: Editable) {
        if (s.length > beforeLength) {
            onTextAdded(s.toString())
        }
    }

    protected abstract fun onTextAdded(newValue: String)
}